gpflow/posterior.py

```diff
@@ -46,19 +63,117 @@ def __init__(self, q_mu, q_sqrt):
         self.q_sqrt = q_sqrt  # [L, M, M], lower-triangular
 
 
-class BasePosterior(Module, ABC):
-    def __init__(self, kernel, inducing_variable, q_mu, q_sqrt, whiten=True, mean_function=None):
+class AbstractPosterior(Module, ABC):
```

Comment:
`AbstractPosterior` doesn't look like an abstract object, it does pretty much lots of things that are not abstract.

Reply:
Abstract in the sense that it's an abstract base class and you can't instantiate it without subclassing & actually implementing the _conditionals methods?